app/models/geolocation.rb

```diff
@@ -0,0 +1,91 @@
+require Rails.root.join("lib/ISO3166/country")
+
+class Geolocation
+  class ArrayType < ActiveRecord::Type::Value
+    include ActiveModel::Type::Helpers::Mutable
```

Comment:
I added a doc comment explaining but in short, since you can directly add/remove elements from an array without swapping out the whole array, we are indicating to ActiveRecord that the type is internally mutable (which slightly changes how the framework treats it).